proposals/NNNN-debug-printf.md

```diff
@@ -0,0 +1,56 @@
+# Dxil Debug Printf
+
+* Proposal: [NNNN](NNNN-debug-printf.md)
+* Author(s): [Jiao Lu](https://github.com/jiaolu)
+* Sponsor: TBD
+* Status: **Under Consideration**
+
+## Introduction
+
+Add new dxil op to allow c/c++ like *printf* intrinsic instructions can be generated in the dxil sourced from hlsl.
+
+## Motivation
+
+As the new shader models, i.e. RayTracing, RayQuery, Workgraph, and more complex algorithms and structure emerges in recent time, we meet many more requirements of debugging capabilities of hlsl coding in game, application and driver development.
+
+The dxil counterpart spirv has a similar feature, [NonSemantic.DebugPrintf extention](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/nonsemantic/NonSemantic.DebugPrintf.asciidoc) to generate DebugPrintf spirvOp souced from hlsl/glsl. Based on the spirvOp extention, some vendor drivers and Valve lunarG has [debug printf layer](https://github.com/KhronosGroup/Vulkan-ValidationLayers/blob/main/docs/debug_printf.md) to dump printf expression, "hlsl/glsl variables" into stdio or file.
+
+## Proposed solution
+
+The printf expression in hlsl mostly like this example.
+```c++ hlsl:
+
+const string str0= "str0";
+string str1 = "str1";
```

Comment:
`string` is part of current semantics,  for example. 

`
const string first = "node";
[Shader(first)]
`

it can be used with the intrinsic which accept strings. but from my experimentation, it can not be used in function argument.

Reply:
It looks like there are restrictions on where strings can appear.  

> printf(str0);

This example seems to be passing a string as a function argument, is this going to be supported now?
 
>  printf("Variables are: %d %d %.2f", 1u, 2u, 1.5f);

This seems to have a string literal being passed as a function argument.

Reply:
The function mentioned is user defined function, but it seem instrinsic functions can use string variable.

Reply:
We don't want to be in a situation where built-in functions behave in a fundamentally different way to user-defined functions.  Where this has been done in the past, this has led to inconsistent behavior with things like overload resolution.  This is something we've been actively trying to address in the implementation of HLSL in Clang, and so any new features we add we want to be sure to add in a way that is compatible with that goal.

Reply:
the string seems touching some history baggage from clang. Now removing the string from example